cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -559,4 +586,21 @@ private String resetToCleanBranch(final String remoteBranchName)
     git.clean().setCleanDirectories(true).setIgnore(true).call();
     return resolveHead().getName();
   }
+
+  /**
+   * Calculates the size of a directory in bytes while ignoring symlinks.
+   *
+   * @param directoryPath the path of the root directory.
+   * @return the size of the directory in bytes.
+   * @throws IOException when there are failures while reading the file.
+   */
+  @VisibleForTesting
+  static long calculateDirectorySize(Path directoryPath) throws IOException {
+    try (Stream<Path> walk = Files.walk(directoryPath.toAbsolutePath())) {
```

Comment:
So we cannot read the size from the root directory, instead have to traverse each files?

Reply:
I checked a few results on Google and it looks like even unix commands traverse the filesystem tree to sum file sizes. [This site] describes the options for getting directory size in Java. I tried using Apache commons, but that tries to follow symbolic links and throws an exception if the links are broken. So I went with this option.